### PR TITLE
Only load all plugins before showin login page

### PR DESCRIPTION
**Pull request by jangernert**

### CODE REVIEW

File: src/Backend/FeedServer.vala [hunk 1]
```diff
@@ -21,8 +21,11 @@ public class FeedReader.FeedServer : GLib.Object {
 	private FeedServerInterface? m_plugin;
 	private Peas.Engine m_engine;
 
+	public signal void PluginsChanedEvent();
```
Comment (brendanlong) [inline src/Backend/FeedServer.vala]: Should be `PluginsChangedEvent`?

File: src/Backend/FeedServer.vala [hunk 2]
```diff
@@ -56,10 +61,32 @@ public class FeedReader.FeedServer : GLib.Object {
...
+	public void LoadAllPlugins()
+	{
+		Logger.debug("FeedServer: load all available plugins");
 		foreach(var plugin in m_engine.get_plugin_list())
 		{
 			m_engine.try_load_plugin(plugin);
 		}
+
+		// have to readd this path, otherwise new icons won't show up
+		Gtk.IconTheme.get_default().add_resource_path("/org/gnome/FeedReader/icons");
```
Comment (brendanlong) [inline src/Backend/FeedServer.vala]: That's odd. Why do we need the icon path to load plugins?

File: src/Widgets/LoginPage.vala [hunk 3]
```diff
@@ -84,6 +72,35 @@ public class FeedReader.LoginPage : Gtk.Stack {
...
+				BackendInfo pluginfo = BackendInfo()
+				{
+					ID = plug.getID(),
+					name = plug.serviceName(),
+					flags = plug.getFlags(),
+					website = plug.getWebsite(),
+					iconName = plug.iconName()
+				};
+				m_accountList.add(new LoginRow(pluginfo));
+			}
```
Comment (brendanlong) [inline src/Widgets/LoginPage.vala]: Should we log if the cast to `FeedServerInterface` fails?

File: src/Backend/FeedServer.vala (hunk 2 above)
Comment (jangernert) [reply src/Backend/FeedServer.vala]: The plugins have logo icons compiled into their binaries as gresource. For some reason these icons can't be found if I don't add this path here again and only broken icons would show up on the login page.

File: src/Widgets/LoginPage.vala (hunk 3 above)
Comment (jangernert) [reply src/Widgets/LoginPage.vala]: At least it wouldn't hurt.

**Pull request merged**